client/my-sites/stats/stats-notices/index.tsx

```diff
@@ -21,6 +21,29 @@ import './style.scss';
 
 const TEAM51_OWNER_ID = 70055110;
 
+const removeStatsPurchaseSuccessParamCalypso = () => {
+	const newUrl = new URL( window.location.href );
+	newUrl.searchParams.delete( 'statsPurchaseSuccess' );
+
+	return window.history.replaceState( null, '', newUrl.toString() );
+};
+
+const removeStatsPurchaseSuccessParamOdyssey = () => {
+	// For Odyssey Stats, we simply update the hash which is equavalent to updating the URL.
+	const currentUrl = new URL( window.location.href );
+	const hashUrl = new URL( window.location.hash.substring( 2 ), window.location.origin );
+	hashUrl.searchParams.delete( 'statsPurchaseSuccess' );
+	currentUrl.hash = `#!${ hashUrl.toString() }`;
+	window.history.replaceState( null, '', currentUrl.toString() );
+};
+
+const removeStatsPurchaseSuccessParam = ( isOdysseyStats: boolean ) => {
+	if ( ! isOdysseyStats ) {
+		return removeStatsPurchaseSuccessParamCalypso();
+	}
+	removeStatsPurchaseSuccessParamOdyssey();
+};
```

Comment:
Could we refactor these functions into their own file(s) and add some simple unit tests?

Reply:
Done in b4875fc